File: app/src/main/java/org/tasks/data/PlaceUsage.java
package org.tasks.data;

import androidx.room.Embedded;
import java.util.Objects;

public class PlaceUsage {
  @Embedded public Place place;
  public int count;

  public Place getPlace() {
    return place;
  }

  public int getColor() {
    return place.getColor();
  }

  public int getIcon() {
    return place.getIcon();
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof PlaceUsage)) {
      return false;
    }
    PlaceUsage that = (PlaceUsage) o;
    return count == that.count && Objects.equals(place, that.place);
  }

  @Override
  public int hashCode() {
    return Objects.hash(place, count);
  }

  @Override
  public String toString() {
    return "PlaceUsage{" + "place=" + place + ", count=" + count + '}';
  }
}

File: data/src/commonMain/kotlin/org/tasks/data/PlaceUsage.kt
package org.tasks.data

import androidx.room.Embedded
import org.tasks.data.entity.Place

class PlaceUsage {
    @Embedded lateinit var place: Place
    var count = 0

    val color: Int
        get() = place.color

    val icon: Int
        get() = place.icon

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is PlaceUsage) return false

        if (place != other.place) return false
        if (count != other.count) return false

        return true
    }

    override fun hashCode(): Int {
        var result = place.hashCode()
        result = 31 * result + count
        return result
    }

    override fun toString(): String = "PlaceUsage(place=$place, count=$count)"
}